### PR TITLE
Pm3.hd changes

### PR DESCRIPTION
**Pull request by BAGAsss**
This is pm3.hd advance development

### CODE REVIEW

**Comment by amet** [comment]
Hi,

I would like to have this all in one commit please, just use -> http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git and remove https://github.com/MrMC/mrmc/pull/26/commits/cb8a224a8b0c9f66be52f5abbf4a586c3e948564 from the commits


then force push into your own branch and changes will reflect here.. once cleaned up, I will pull it in

File: addons/skin.pm3.hd/720p/Home.xml [hunk 1]
```diff
@@ -735,7 +780,6 @@
 			</control>
 			<control type="button" id="20">
 				<description>Power push button</description>
-				<visible>!System.Platform.IOS + !System.Platform.TVOS</visible>
```
Comment (amet) [inline addons/skin.pm3.hd/720p/Home.xml]: this cant be removed, has to stay in

File: addons/skin.pm3.hd/720p/Home.xml [hunk 2]
```diff
@@ -599,17 +599,47 @@
 						<onclick>ActivateWindow(Videos,MovieTitles,return)</onclick>
 						<icon>special://skin/backgrounds/movies.jpg</icon>
 						<thumb>$INFO[Skin.String(Home_Custom_Back_Movies_Folder)]</thumb>
-						<visible>Library.HasContent(Movies) | Library.HasServices</visible>
```
Comment (amet) [inline addons/skin.pm3.hd/720p/Home.xml]: why this change???

File: addons/skin.pm3.hd/720p/Home.xml [hunk 3]
```diff
@@ -599,17 +599,47 @@
...
+						<visible>Library.HasContent(Movies)</visible>
 						<visible>!Skin.HasSetting(homepageHideMoviesButton)</visible>
 					</item>
 					<item id="10">
 						<label>20343</label>
 						<onclick>ActivateWindow(Videos,TVShowTitles,return)</onclick>
 						<icon>special://skin/backgrounds/tvshow.jpg</icon>
 						<thumb>$INFO[Skin.String(Home_Custom_Back_TVShow_Folder)]</thumb>
-						<visible>Library.HasContent(TVShows) | Library.HasServices</visible>
```
Comment (amet) [inline addons/skin.pm3.hd/720p/Home.xml]: same question, this needs to stay in... plex and emby are Services

File: addons/skin.pm3.hd/720p/MyPVRChannels.xml [hunk 4]
```diff
@@ -134,6 +184,7 @@
 					<texture>separator.png</texture>
 					<visible>Container(50).ListItem.HasEpg</visible>
 				</control>
+
```
Comment (amet) [inline addons/skin.pm3.hd/720p/MyPVRChannels.xml]: unnecessary white lines

File: addons/skin.pm3.hd/720p/SkinSettings.xml [hunk 5]
```diff
@@ -373,6 +373,199 @@
 					<texturenofocus>-</texturenofocus>
 					<textureradioon>radiobutton-focus-2.png</textureradioon>
 				</control>
+<!-- -->
```
Comment (amet) [inline addons/skin.pm3.hd/720p/SkinSettings.xml]: does nothing, remove please

File: addons/skin.pm3.hd/720p/SkinSettings.xml [hunk 6]
```diff
@@ -559,6 +752,40 @@
 								<visible>Library.HasContent(TVShows)</visible>
 								<visible>!Skin.HasSetting(homepageHideTVShowButton)</visible>
 							</item>
+<!--	-->
```
Comment (amet) [inline addons/skin.pm3.hd/720p/SkinSettings.xml]: same as above

File: addons/skin.pm3.hd/720p/SkinSettings.xml [hunk 7]
```diff
@@ -589,7 +816,7 @@
 								<onclick>noop</onclick>
 								<icon>special://skin/backgrounds/weather.jpg</icon>
 								<thumb>$INFO[Skin.String(Home_Custom_Back_Weather_Folder)]</thumb>
-								<visible>!Skin.HasSetting(homepageHideWeatherButton)</visible>
+								<visible>!Skin.HasSetting(homepageHideWeatherButton) + !IsEmpty(Weather.Plugin)</visible>
```
Comment (amet) [inline addons/skin.pm3.hd/720p/SkinSettings.xml]: what weather plugin, we dont have weather...

File: addons/skin.pm3.hd/720p/Font.xml [hunk 8]
```diff
@@ -1,181 +1,469 @@
...
+	 	<font>
+			<name>font35b</name>
+			<filename>LiberationSans-Regular.ttf</filename>
+			<size>82</size>
+			<aspect>.8</aspect>
+		</font>
+
+<!-- Addon fonts -->
+
+<font>
```
Comment (amet) [inline addons/skin.pm3.hd/720p/Font.xml]: none of the osmc_addon_* fonts are being used.... why is it in?

File: addons/skin.pm3.hd/720p/Font.xml [hunk 9]
```diff
@@ -1,181 +1,469 @@
...
+			<size>45</size>
+		</font>
+	 	<font>
 			<name>font35b</name>
 			<filename>arial.ttf</filename>
 			<size>80</size>
 			<aspect>.8</aspect>
 		</font>
 	</fontset>
-	<fontset id="Unicode" unicode="true">
```
Comment (amet) [inline addons/skin.pm3.hd/720p/Font.xml]: you cant remove unifont-9.0.06.ttf fonts, I have added them in for a reason

**Comment by amet** [comment]
phase 1 of the review id done, now squash it all in into one commit and I'll look it over again

remove https://github.com/MrMC/mrmc/pull/26/commits/cb8a224a8b0c9f66be52f5abbf4a586c3e948564 please

**Comment by BAGAsss** [comment]
Hi!

Sorry but i cannot understood what u asked me to do - i new one in github

in my branch i already made all changes more than month ago.

PS - all questions in your later e-mails - it add possibility use 4 
smart-playlists on homescreen cause tv-shows shows all series as movies 
and cartoons as movies section too

here u can make up to 4 smart-playlist as u want and put it on 
homescreen for quick call.

Other changes - corrections for better use.

File: addons/skin.pm3.hd/720p/Font.xml (hunk 9 above)
Comment (BAGAsss) [inline addons/skin.pm3.hd/720p/Font.xml]: I added 2 section fonsets Arial Mid and Arial Big - for blind man like me - it is only increase letters size and put it in top of file. While GitHub desided i try to delete some lines - i don`t know ((((

File: addons/skin.pm3.hd/720p/Font.xml (hunk 8 above)
Comment (BAGAsss) [inline addons/skin.pm3.hd/720p/Font.xml]: sorry, now i cannot say which version of Kodi pm3.hd i use to develop Font.xml - i think from 16. i`ve added couple <fontset> Arial group - no more.

Comment (amet) [inline addons/skin.pm3.hd/720p/Font.xml]: I understand, but in this PR you are proposing we add this in... you need to be developing on top of MrMC master if you would like this PR to go in

File: addons/skin.pm3.hd/720p/Font.xml (hunk 9 above)
Comment (amet) [inline addons/skin.pm3.hd/720p/Font.xml]: see my comment above, you have to be working with addons/skin.pm3.hd/720p/Font.xml from MrMC as that has all we need in it + you add your changes

**Comment by amet** [comment]
changes in your branch work for you as you are not interested in unifont-9.0.06.ttf for example, what I have asked is for whitelines and unnecessary changes to be removed and that functionality not be changed for some items, like quit button has to stay hidden on tvos and ios and <visible> tags to include HasServices, unifont to not be removed ... etc

File: addons/skin.pm3.hd/720p/Font.xml (hunk 9 above)
Comment (BAGAsss) [inline addons/skin.pm3.hd/720p/Font.xml]: i`ll recode it in couple hours

**Comment by BAGAsss** [comment]
I tried do not remove additional lines and comment it for future using.

**Comment by amet** [comment]
that doesnt make sense... show me which line? `<!-- -->` is just a ugly line that does nothing ... anything between `<!--` and `-->`is commented out, but there is nothing between that in your lines https://github.com/MrMC/mrmc/pull/26/files/cb8a224a8b0c9f66be52f5abbf4a586c3e948564#diff-f1d2876f079b17fe5192adc95d59dce6R376

**Comment by amet** [comment]
please understand that we will not include code just because you asked for it, you are pushing for your changes to go in and we will gladly accept it but only under strict coding guidelines, we have to make sure that code is readable and understandable in future. this can only go in as one commit, so I suggest you start reading up on how to squash all these changes into one commit and force push back into your branch

**Comment by BAGAsss** [comment]
I understand u spend your time to explain me all of this.
If your colleague did it early we can put off a lot of questions this 
moment.
i`ll go to read docs how to squash and ......

**Comment by amet** [comment]
unfortunately we had and still have more important issues than a skin that very few people use, we appreciate that you are trying to help and that these changes will be useful to others. point is that video playback and bugs always take priority over a skin enhancements

File: addons/skin.pm3.hd/720p/Home.xml [hunk 10]
```diff
@@ -671,13 +671,15 @@
 						<thumb>$INFO[Skin.String(Home_Custom_Back_Programs_Folder)]</thumb>
 						<visible>!Skin.HasSetting(homepageHideProgramButton)</visible>
 					</item>
+<!--
 					<item id="5">
 						<label>8</label>
 						<onclick>ActivateWindow(Weather)</onclick>
 						<icon>special://skin/backgrounds/weather.jpg</icon>
 						<thumb>$INFO[Skin.String(Home_Custom_Back_Weather_Folder)]</thumb>
 						<visible>!Skin.HasSetting(homepageHideWeatherButton) + !IsEmpty(Weather.Plugin)</visible>
```
Comment (amet) [inline addons/skin.pm3.hd/720p/Home.xml]: once you squash this into one commit, I will still be complaining about this line :)

**Comment by BAGAsss** [comment]
I understood - but u find a little time to explain

**Comment by amet** [comment]
I dont understand what you mean, there is tons of information on how git(and github) works, should not be difficult to google it. I have accepted to review this and accepted that your changes will go in once code guidelines are followed

**Comment by BAGAsss** [comment]
i made:
1. installed git
2. clone https://github.com/BAGAsss/mrmc.git
3. git rebase -i HEAD~10 and delete all exclude my changes and replace 
pick to squash
4. git push --force
5. git reset --hard cb8a224a8b0c9f66be52f5abbf4a586c3e948564
6. git push --force
Is it right? is it enough to get corrected pm3hd in 3.0.5?
Thanks in advance

**Comment by amet** [comment]
This pr is still showing 26 commits... did you push into your github?

**Comment by BAGAsss** [comment]
git push --force
it request login and password and.....

Username for 'https://github.com': BAGAsss
Password for 'https://BAGAsss@github.com':
Counting objects: 25, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (15/15), done.
Writing objects: 100% (25/25), 1.95 MiB | 661.00 KiB/s, done.
Total 25 (delta 17), reused 15 (delta 8)
remote: Resolving deltas: 100% (17/17), completed with 17 local objects.
To https://github.com/BAGAsss/mrmc.git
  + 87c2c6a6d7...fcb2bb671a master -> master (forced update)

**Comment by amet** [comment]
It's not working, https://github.com/BAGAsss/mrmc/commit/cb8a224a8b0c9f66be52f5abbf4a586c3e948564

**Comment by BAGAsss** [comment]
what i should do?

**Comment by amet** [comment]
You should clean it up, it can't go in like this

**Comment by BAGAsss** [comment]
i can drop repo (i think) and fork new one.
